### PR TITLE
Pr nimrod bug186

### PR DESCRIPTION
**Pull request by NimrodGolan**
### References
https://dev.azure.com/CxFlow/CxFlow/_workitems/edit/186

### CODE REVIEW

**Comment by cxflowtestuser** [comment]
Scan submitted to Checkmarx

**Comment by cxflowtestuser** [comment]
### Checkmarx scan completed
[Full Scan Details](http://EC2AMAZ-JP2I9QF/CxWebClient/ViewerMain.aspx?scanid=1010484&projectid=164)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|112
Low|1401
Informational|56

#### Violation Summary
Severity|Count
---|---

#### Details
|Lines|Severity|Category|File|Link|
---|---|---|---|---

File: src/main/java/com/checkmarx/flow/service/ThresholdValidatorImpl.java [hunk 1]
```diff
@@ -34,6 +32,17 @@
     
     private final FlowProperties flowProperties;
     private final ScaProperties scaProperties;
+    private final SastScanner sastScanner;
+    private final SCAScanner scaScanner;
+
+    @Autowired
+    public ThresholdValidatorImpl(@Lazy SastScanner sastScanner, @Lazy SCAScanner scaScanner,
```
Comment (ofersk) [inline src/main/java/com/checkmarx/flow/service/ThresholdValidatorImpl.java:39]: why do you need a constructor?  put '@autowired' on the fields

File: src/main/java/com/checkmarx/flow/service/ThresholdValidatorImpl.java [hunk 2]
```diff
@@ -61,16 +70,15 @@ private boolean isAllowed(ScanResults scanResults, RepoProperties repoProperties
             isAllowed = isAllowedSast(scanResults, pullRequestReport);
         }
 
-        if (isAllowed && StringUtils.containsIgnoreCase(flowProperties.getEnabledVulnerabilityScanners().toString(), ScaProperties.CONFIG_PREFIX)) {
+        if (isAllowed && scaScanner.isEnabled()) {
             isAllowed = isAllowedSca(scanResults, pullRequestReport);
```
Comment (ofersk) [inline src/main/java/com/checkmarx/flow/service/ThresholdValidatorImpl.java:74]: we can put the isenables inside the isAllowed (I think they always go together), then this function will be:
`return SastScanner.isAllowed() && scaScanner.isAllowed()`

File: src/main/java/com/checkmarx/flow/service/ThresholdValidatorImpl.java [hunk 3]
```diff
@@ -61,16 +70,15 @@ private boolean isAllowed(ScanResults scanResults, RepoProperties repoProperties
...
         }
 
         return isAllowed;
     }
 
     private boolean isSast() {
```
Comment (ofersk) [inline src/main/java/com/checkmarx/flow/service/ThresholdValidatorImpl.java:80]: inline if posible

**Pull request merged**